runtime/src/rpc.rs

```diff
@@ -663,24 +790,54 @@ impl UtilFuncs for InterBtcParachain {
         &vault_id.account_id == self.get_account_id()
     }
 
-    async fn get_foreign_assets_metadata(&self) -> Result<Vec<(u32, AssetMetadata)>, Error> {
+    async fn get_decoded_storage_keys<T, U, F>(
```

Comment:
I don't think this should be a trait function - I think this should be a private function in the `impl InterBtcParachain`, just like other internal utility functions like `with_unique_signer`